lib/addons/vpc-cni/index.ts

```diff
@@ -1,15 +1,241 @@
+  */
+  awsVpcK8sCniLogFile?: string;
+  /**
+  * `AWS_VPC_K8S_CNI_RANDOMIZESNAT` Environment Variable. Type: String.
+  * Specifies whether the SNAT iptables rule should randomize the outgoing 
+  * ports for connections.
+  */
+  awsVpcK8sCniRandomizeSnat?: string;
+  /**
+  * `AWS_VPC_K8S_CNI_EXTERNALSNAT` Environment Variable. Type: Boolean.
+  * Specifies whether an external NAT gateway should be used to provide SNAT 
+  * of secondary ENI IP addresses.
+  */
+  awsVpcK8sCniExternalSnat?: boolean;
+  /**
+  * `AWS_VPC_K8S_PLUGIN_LOG_FILE` Environment Variable. Type: String.
+  * Specifies where to write the logging output for aws-cni plugin. 
+  */
+  awsVpcK8sPluginLogFile?: string;
+  /**
+  * `AWS_VPC_K8S_PLUGIN_LOG_LEVEL` Environment Variable. Type: String.
+  * Specifies the loglevel for aws-cni plugin.
+  */
+  awsVpcK8sPluginLogLevel?: string;
+  /**
+  * `DISABLE_INTROSPECTION` Environment Variable. Type: Boolean.
+  */
+  disableIntrospection?: boolean;
+  /**
+  * `DISABLE_METRICS` Environment Variable. Type: Boolean.
+  * Specifies whether the prometheus metrics endpoint is disabled or not for 
+  * ipamd. By default metrics are published on :61678/metrics.
+  */
+  disableMetrics?: boolean;
+  /**
+  * `DISABLE_NETWORK_RESOURCE_PROVISIONING` Environment Variable. Type: Boolean.
+  * Setting DISABLE_NETWORK_RESOURCE_PROVISIONING to true will make IPAMD 
+  * depend only on IMDS to get attached ENIs and IPs/prefixes.
+  */
+  disablenetworkResourceProvisioning?: boolean;
+  /**
+  * `ENABLE_POD_ENI` Environment Variable. Type: Boolean.
+  * Setting ENABLE_POD_ENI to true will allow IPAMD to add the 
+  * vpc.amazonaws.com/has-trunk-attached label to the node if the instance 
+  * has the capacity to attach an additional ENI.
+  */
+  enablePodEni?: boolean;
+  /**
+  * `ENABLE_PREFIX_DELEGATION` Environment Variable. Type: Boolean.
+  * To enable prefix delegation on nitro instances. Setting 
+  * ENABLE_PREFIX_DELEGATION to true will start allocating a prefix (/28 for IPv4 and /80 for IPv6) instead of a secondary IP in the ENIs subnet. 
+  */
+  enablePrefixDelegation?: boolean;
+  /**
+  * `WARM_ENI_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free elastic network interfaces (and all of their 
+  * available IP addresses) that the ipamd daemon should attempt to keep 
+  * available for pod assignment on the node. 
+  */
+  warmEniTarget?: number;
+  /**
+  * `WARM_PREFIX_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free IPv4(/28) prefixes that the ipamd daemon 
+  * should attempt to keep available for pod assignment on the node.
+  */
+  warmPrefixTarget?: number;
+  /**
+   * Secondary Subnet IDs for creating `ENIConfig`
+   */
+  subnetIds?: string[];
+  /**
+   * Corrosponding Availability Zones of the Secondary Subnet IDs 
+   * for creating `ENIConfig`
+   */
+  availabilityZones?: string[];
```

Comment:
Are these redundant? Would not subnets already be part of a particular az? Can we derive it from the subnets?